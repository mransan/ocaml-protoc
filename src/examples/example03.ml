let () = 
  let v = Example03_types.Some "This is an example" in  
  Format.fprintf Format.std_formatter "%a\n" Example03_pp.pp_string_some v
