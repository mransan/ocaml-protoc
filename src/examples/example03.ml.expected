[@@@ocaml.warning "-27-30-39"]

type string_some_none = unit

type string_some_t =
  | None
  | Some of string

and string_some = {
  t : string_some_t option;
}

let rec default_string_some_none = ()

let rec default_string_some_t (): string_some_t = None

and default_string_some 
  ?t:((t:string_some_t option) = None)
  () : string_some  = {
  t;
}

[@@@ocaml.warning "-27-30-39"]

(** {2 Formatters} *)

let rec pp_string_some_none fmt (v:string_some_none) = 
  let pp_i fmt () =
    Pbrt.Pp.pp_unit fmt ()
  in
  Pbrt.Pp.pp_brk pp_i fmt ()

let rec pp_string_some_t fmt (v:string_some_t) =
  match v with
  | None  -> Format.fprintf fmt "None"
  | Some x -> Format.fprintf fmt "@[<hv2>Some(@,%a)@]" Pbrt.Pp.pp_string x

and pp_string_some fmt (v:string_some) = 
  let pp_i fmt () =
    Pbrt.Pp.pp_record_field ~first:true "t" (Pbrt.Pp.pp_option pp_string_some_t) fmt v.t;
  in
  Pbrt.Pp.pp_brk pp_i fmt ()
